### PR TITLE
Ml separate dev

### PR DESCRIPTION
**Pull request by OscarVlld**
- Create separate class LicchaviDev(Licchavi)
- Create new command "ml_train_dev"
- Remove test options from Licchavi()
- Remove dev imports from prodution code

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -8,4 +8,5 @@ python_files = tests.py test_*.py *_tests.py
 filterwarnings =
     # TODO: could fix the issues instead of ignoring the warning here: 
     # https://github.com/audiolion/django-language-field 
-    ignore:django.utils.translation.ugettext_lazy\(\) is deprecated:DeprecationWarning
\ No newline at end of file
+    ignore:django.utils.translation.ugettext_lazy\(\) is deprecated:DeprecationWarning
+    ignore::django.utils.deprecation.RemovedInDjango41Warning
```
Comment (OscarVlld) [inline pytest.ini:12]: Same as my other PR, I don't know if we want this on main.

Comment (lfaucon) [reply pytest.ini:12]: Yes, we will update this on main because we can't do anything about this warning.

File: ml/dev/licchavi_dev.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    nb_loc += 1
+            loc_mean_error = loc_error / nb_loc
+        return glob_mean_error, loc_mean_error
+
+    def _update_hist(self, epoch, fit, s, gen, reg):
+        """ Updates history (at end of epoch)
+
+        Inheriting from Licchavi._update_hist()
+         """
+        super()._update_hist(epoch, fit, s, gen, reg)
```
Comment (lfaucon) [inline ml/dev/licchavi_dev.py:82]: 👍

File: ml/management/commands/ml_train_dev.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+from django.core.management.base import BaseCommand
+
+from ml.core import TOURNESOL_DEV
+from .ml_train import fetch_data
+from ml.dev.experiments import run_experiment
+
+
+"""
+Machine Learning main python file
```
Comment (lfaucon) [inline ml/management/commands/ml_train_dev.py:11]: Probably better to not call it "main python file", but "python file for development"

**Review by lfaucon — CHANGES_REQUESTED** [review]

**Comment by lfaucon** [comment]
Nice refactor!

File: ml/management/commands/ml_train_dev.py (hunk 3 above)
Comment (OscarVlld) [reply ml/management/commands/ml_train_dev.py:11]: Yes, thanks. I forgot to change it after copy-pasting ml_train.

File: ml/dev/licchavi_dev.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        vid_vidx (dictionnary): dictionnary of {video ID: video index}
+        crit (str): comparison criteria learnt
+        test_mode (bool): wether we use fake data or not
+        device (str): device used (cpu/gpu)
+        verb (float): verbosity level
+        """
+        # Inits Licchavi class
+        super().__init__(nb_vids, vid_vidx, crit, device=device, verb=verb)
+
+        self.test_mode = test_mode
```
[the remark this answers is not in the document]
Comment (OscarVlld) [reply ml/dev/licchavi_dev.py:30]: It is actually like this:
- Licchavi -> test_mode == False  (so I removed it altogether)
- LicchaviDev -> test_mode == True or False  (so I kept it)

Indeed, self.test_mode is boolean for the usage of artificially generated data (providing ground truths). This type of data is never used in production, but not always for developpement (I sometimes use real data or manual comparisons as input). As I plan to use only LicchaviDev for tests, it must be able to adapt to those different situations.

[the remark this answers is not in the document]
Comment (lfaucon) [reply ml/dev/licchavi_dev.py:30]: Understood! Thanks for clarifying

**Pull request merged**